## MOVIES DATASET ANALYSIS

### OBJECTIVE :

Maximize financial performance in the film industry by investing in successful genres, utilizing renowned actors and directors, and strategically planning release dates. Increase profitability through remakes and franchise opportunities, nurturing new talent, and replicating successful strategies while minimizing financial risks.

#### Connect database with jupyter notebook / python

In [1]:
# Import the libraries
import mysql.connector as sql #To establish a connection to the MySQL database.
import pandas as pd #To work on dataframe.

#### Establish a connection to the MySQL database

In [3]:
mydb = sql.connect(host='localhost',  # Specifies the hostname or IP address of the MySQL server.
                   user='root',       # Specifies the username to connect to the database.
                   passwd='pass@123', # Specifies the password for the user 'root'.
                   use_pure=True,     # Indicates whether to use the pure Python implementation of MySQL Connector/Python.
                   database='etlhive') # Specifies the name of the database you want to connect to.
print(mydb)


In [4]:
# Used to manage Python warnings
from warnings import filterwarnings
filterwarnings('ignore')

#### ACCESS TABLE FROM MYSQL

In [5]:
data = pd.read_sql_query('select * from movies',mydb)
data

Movie_Name Release_Period Whether_Remake  \
0                           Golden Boys         Normal             No   
1                         Kaccha Limboo        Holiday             No   
2                      Not A Love Story        Holiday             No   
3                            Qaidi Band        Holiday             No   
4                             Chaatwali        Holiday             No   
...                                 ...            ...            ...   
1667                         Fight Club        Holiday             No   
1668                 Strings Of Paasion         Normal             No   
1669              Dunno Y Na Jaane Kyun         Normal             No   
1670  Taj Mahal - An Eternal Love Story         Normal             No   
1671                   Mr. Hot Mr. Kool         Normal             No   

     Whether_Franchise     Genre New_Actor New_Director New_Music_Director  \
0                   No  suspense       Yes           No                 No   
1                   No     drama       Yes           No                Yes   
2                   No  thriller        No           No                 No   
3                   No     drama       Yes           No                 No   
4                   No     adult       Yes          Yes                Yes   
...                ...       ...       ...          ...                ...   
1667                No    action        No          Yes                 No   
1668                No     drama        No          Yes                Yes   
1669                No     drama        No           No                 No   
1670                No     drama        No          Yes                 No   
1671                No    comedy        No           No                Yes   

             Lead_Star               Director   Music_Director  \
0         Jeet Goswami             Ravi Varma    Baba Jagirdar   
1     Karan Bhanushali          Sagar Ballary  Amardeep Nijjer   
2           Mahie Gill        Ram Gopal Verma   Sandeep Chowta   
3           Aadar Jain           Habib Faisal     Amit Trivedi   
4           Aadil Khan             Aadil Khan     Babloo Ustad   
...                ...                    ...              ...   
1667        Zayed Khan          Vikram Chopra           Pritam   
1668       Zeenat Aman  Sanghamitra Chaudhuri       Dev Sikdar   
1669       Zeenat Aman          Sanjay Sharma           Nikhil   
1670       Zulfi Sayed             Akbar Khan          Naushad   
1671       Zulfi Sayed           Partho Ghosh   Rishi - Ranjit   

      Number_of_Screens     Budget   Revenue  
0                     5    5000000     85000  
1                    75   15000000    825000  
2                   525   75000000  56700000  
3                   800  210000000   4500000  
4                     1    1000000   1075000  
...                 ...        ...       ...  
1667                375   82500000  88862500  
1668                 10    8000000     70000  
1669                 20   12500000    850000  
1670                135  100000000  31065000  
1671                 30   27500000   1300000  

[1672 rows x 14 columns]

#### PERFORMING SQL QUERIES TO ANALYSE DATASET

### 1. BASIC DESCRIPTIVE STATISTICS

###### 1. Count the total number of movies.

In [6]:
df = pd.read_sql_query(" select count(*) as total_movies from movies ",mydb)
df

total_movies
0          1672

###### 2. Get the total and average revenue of all movies

In [7]:
df = pd.read_sql_query(" select sum(revenue) as total_revenue, avg(revenue) as average_revenue from movies ",mydb)
df

total_revenue  average_revenue
0   3.029960e+11     1.812177e+08

###### 3. Get the total and average budget of all movies

In [8]:
df = pd.read_sql_query(" select sum(budget) as total_budget, avg(budget) as average_budget from movies ",mydb)
df

total_budget  average_budget
0  2.229442e+11    1.333398e+08

The 1,672 movies generated a total revenue of  303 billion dollars with an average revenue of  181 million dollars and a total budget of 223 billion dollars with an average budget of 133 million dollars.This indicates a healthy financial performance in the movie industry, with significant investments yielding substantial returns.

### 2. GENRE-BASED INSIGHTS

###### 4.Count the number of movies by genre

In [9]:
df = pd.read_sql_query(" select genre,count(*) as movie_count from movies1 group by genre order by movie_count desc ",mydb)
df

genre  movie_count
0          drama          634
1         comedy          283
2       thriller          212
3     love_story          131
4         action          116
5       rom__com           94
6          adult           78
7         horror           53
8       suspense           30
9   mythological           14
10       fantasy           13
11        masala           10
12     animation            3
13   documentary            1

###### 5. Find the average budget and revenue for each genre

In [10]:
df = pd.read_sql_query(" select genre,avg(budget) as average_budget, avg(revenue) as average_revenue from movies group by genre order by average_revenue desc",mydb)
df

genre  average_budget  average_revenue
0         masala    7.090000e+08     1.200703e+09
1    documentary    3.900000e+08     6.495150e+08
2         action    2.667823e+08     3.680839e+08
3       rom__com    1.972947e+08     2.722939e+08
4         comedy    1.482350e+08     2.242688e+08
5     love_story    1.435840e+08     2.204200e+08
6       thriller    1.558420e+08     1.951393e+08
7          drama    1.008156e+08     1.278382e+08
8         horror    7.110377e+07     9.700698e+07
9        fantasy    1.996154e+08     8.581202e+07
10      suspense    3.875000e+07     2.062833e+07
11     animation    9.750000e+07     1.330000e+07
12         adult    3.957692e+06     4.591378e+06
13  mythological    9.214286e+06     7.280357e+05

###### 6. Find the highest-grossing movie in each genre

In [14]:
df = pd.read_sql_query(" select m.genre, m.movie_name, m.revenue from movies m inner join ( select genre, max(revenue) as max_revenue from movies group by genre) as max_rev on m.genre = max_rev.genre and m.revenue = max_rev.max_revenue order by m.revenue desc ",mydb)
df

genre                 movie_name     revenue
0     love_story           Jab Tak Hai Jaan  2102642500
1          drama                  Tubelight  2079900000
2         action                     Ra.One  2073862500
3         masala                   Judwaa 2  2066477500
4         comedy                Housefull 3  1949350000
5       thriller                        Fan  1849330000
6       rom__com        Jab Harry Met Sejal  1477580000
7         horror                     Raaz 3   952662500
8    documentary  Sachin - A Billion Dreams   649515000
9       suspense              36 China Town   379200000
10       fantasy                      Joker   321087500
11         adult        Love Sex Aur Dhokha   115200000
12     animation                      Jumbo    34700000
13  mythological           Jai Santoshi Maa     3100000

Genre Drama has the highest no. of movies with 634 count, while Masala genre shows highest average revenue (1.2B Dollars) but low movie count. Love Story's "Jab Tak Hai Jaan" being the highest-grossing movie(2.10B Dollars) overall,highlighting the strong performance of various genres and their potential for blockbuster success.

### 3. Star and Director Insights

###### 7. Find the number of movies each lead star has acted in

In [15]:
df = pd.read_sql_query(" select lead_star,count(*) as total_movies from movies1 group by lead_star order by total_movies desc",mydb)
df

lead_star  total_movies
0        Akshay Kumar            48
1          Ajay Devgn            33
2       Emraan Hashmi            27
3    Amitabh Bachchan            21
4       Saif Ali Khan            18
..                ...           ...
759   Yugboah Garewal             1
760       Yugraj Jain             1
761     Yukta Mookhey             1
762       Zaira Wasim             1
763     Zakir Hussain             1

[764 rows x 2 columns]

###### 8. Find the number of movies directed by each director

In [16]:
df = pd.read_sql_query(" select director,count(*) as total_movies from movies1 group by director order by total_movies desc",mydb)
df

director  total_movies
0           Ram Gopal Verma            18
1              Vikram Bhatt            16
2              Priyadarshan            16
3               Suresh Jain            15
4                Mohit Suri            11
...                     ...           ...
1040           Sahil Sangha             1
1041         Suresh Krishna             1
1042          Vikram Chopra             1
1043  Sanghamitra Chaudhuri             1
1044             Akbar Khan             1

[1045 rows x 2 columns]

###### 9. Find the average revenue of movies for each lead star

In [17]:
df = pd.read_sql_query(" select lead_star,avg(revenue) as average_revenue from movies group by lead_star order by average_revenue desc ",mydb)
df

lead_star  average_revenue
0            Prabhas     1.682350e+09
1      Shahrukh Khan     1.505104e+09
2         Alia Bhatt     1.389138e+09
3       Varun Dhawan     1.320808e+09
4     Hrithik Roshan     1.258926e+09
..               ...              ...
759  Umed Khandelwal     3.250000e+04
760  Abhishek Sharma     1.600000e+04
761   Nishant Sharma     1.450000e+04
762             Tina     1.200000e+04
763   Yatin Karyekar     7.250000e+03

[764 rows x 2 columns]

###### 10. Find the average budget of movies for each director

In [18]:
df = pd.read_sql_query(" select director,avg(budget) as average_budget from movies group by director order by average_budget desc ",mydb)
df

director  average_budget
0       S.S. Rajamouli    1.700000e+09
1       Karan Malhotra    9.150000e+08
2       Farhad - Sajid    8.900000e+08
3      Abhinav Kashyap    8.300000e+08
4                Krish    7.900000e+08
...                ...             ...
1040       Vinay Arora    1.000000e+06
1041     Swapnil Koche    7.500000e+05
1042          Jyotiish    5.000000e+05
1043  Ravi Ved Chouhan    5.000000e+05
1044   Mahesh D. Singh    3.250000e+05

[1045 rows x 2 columns]

Akshay Kumar leads with 48 movies, Ram Gopal Verma directed 18 movies with highest count, Prabhas tops in average revenue (1.682B Dollars) among lead stars, while S.S. Rajamouli leads with the highest average budget (1.7B Dollars) among directors, reflecting their significant impact on movie success and investment.

### 4. Release Period Insights

###### 12. Count the number of movies released in each period

In [19]:
df = pd.read_sql_query(" select release_period, count(*) from movies group by release_period ",mydb)
df

release_period  count(*)
0         Normal      1058
1        Holiday       614

###### 13. Find the total revenue for movies released during holidays vs. non-holidays

In [20]:
df = pd.read_sql_query("select release_period,sum(revenue) as total_revenue from movies group by release_period ",mydb)
df

release_period  total_revenue
0         Normal   1.614953e+11
1        Holiday   1.415007e+11

The highest number of movies were released in the normal period, with total revenues of 141B Dollars during holidays and 161B Dollars during non-holidays.

### 5. Remake and Franchise Insights

###### 14. Count the number of remakes and franchises

In [21]:
df = pd.read_sql_query(" select whether_remake, whether_franchise, count(*) as movie_count from movies group by whether_remake, whether_franchise",mydb)
df

whether_remake whether_franchise  movie_count
0             No                No         1530
1            Yes                No           68
2             No               Yes           74

###### 15. Find the average revenue of remakes vs. non-remakes

In [22]:
df = pd.read_sql_query(" select Whether_Remake,avg(revenue) as average_revenue from movies group by Whether_Remake ",mydb)
df

Whether_Remake  average_revenue
0             No     1.657120e+08
1            Yes     5.469681e+08

###### 16. Find the average revenue of franchise movies vs. non-franchise movies

In [23]:
df = pd.read_sql_query(" select whether_franchise, avg(revenue) as average_revenue from movies group by whether_franchise",mydb)
df

whether_franchise  average_revenue
0                No     1.604267e+08
1               Yes     6.301896e+08

The dataset contains 1,530 movies without remakes or franchises, 68 with remakes, and 74 with franchises. Remakes have an average revenue of 546M Dollars, non-remakes 165M Dollars, franchises 630M Dollars, and non-franchises 160M Dollars.

### 6. New Talent Insights

###### 17. Count the number of movies with new actors, directors, and music directors

In [24]:
df = pd.read_sql_query(" select new_actor, new_director, new_music_director, count(*) as movie_count from movies group by new_actor, new_director, new_music_director order by movie_count desc ",mydb)
df

new_actor new_director new_music_director  movie_count
0        No           No                 No          585
1        No          Yes                 No          326
2       Yes          Yes                Yes          186
3        No          Yes                Yes          172
4        No           No                Yes          134
5       Yes          Yes                 No          128
6       Yes           No                 No           83
7       Yes           No                Yes           58

###### 18. Find the average budget and revenue for movies with new actors

In [25]:
df = pd.read_sql_query(" select new_actor, avg(budget) as average_budget, avg(revenue) as average_revenue from movies group by new_actor;",mydb)
df

new_actor  average_budget  average_revenue
0       Yes    4.967560e+07     4.267826e+07
1        No    1.646194e+08     2.330134e+08

###### 19. Find the average budget and revenue for movies with new directors

In [26]:
df = pd.read_sql_query(" select new_director, avg(budget) as average_budget, avg(revenue) as average_revenue from movies group by new_director;",mydb)
df

new_director  average_budget  average_revenue
0           No    1.914735e+08     2.800315e+08
1          Yes    7.176974e+07     7.656262e+07

###### 20. Find the average budget and revenue for movies with new music directors

In [27]:
df = pd.read_sql_query(" select new_music_director, avg(budget) as average_budget, avg(revenue) as average_revenue from movies group by new_music_director;",mydb)
df

new_music_director  average_budget  average_revenue
0                 No    1.685217e+08     2.342711e+08
1                Yes    6.156877e+07     7.298863e+07

The majority of movies feature established actors, directors, and music directors. Movies that introduce new talent in these roles tend to have lower budgets and generate lower revenues compared to movies with experienced talent.

### 7. Performance Insights

###### 21. Retrieve the top 10 highest-grossing movies

In [28]:
df = pd.read_sql_query(" select movie_name,revenue from movies order by revenue desc limit 10 ",mydb)
df

movie_name     revenue
0                  Jab Tak Hai Jaan  2102642500
1                         Tubelight  2079900000
2                            Ra.One  2073862500
3                          Judwaa 2  2066477500
4                             Don 2  2028100000
5  Goliyon Ki Raasleela Ram - Leela  2014805000
6              Toilet Ek Prem Katha  2000800000
7                     Rowdy Rathore  1984775000
8                       Housefull 3  1949350000
9             Badrinath Ki Dulhania  1948152500

###### 22. Retrieve the top 10 most expensive movies by budget

In [29]:
df = pd.read_sql_query(" select movie_name, budget from movies order by budget desc limit 10 ",mydb)
df

movie_name      budget
0       Bahubali - The Beginning  1700000000
1                   Mohenjo Daro  1380000000
2                      Tubelight  1350000000
3                   Jagga Jasoos  1310000000
4                         Ra.One  1300000000
5                            Fan  1200000000
6            Jab Harry Met Sejal  1190000000
7                  Bombay Velvet  1180000000
8                       Brothers  1120000000
9  M.S. Dhoni - The Untold Story  1100000000

###### 23. Find the top 5 most profitable movies (profit = revenue - budget)

In [30]:
df = pd.read_sql_query(" select movie_name,revenue-budget as profit from movies order by profit desc limit 5",mydb)
df

movie_name      profit
0  Badrinath Ki Dulhania  1508152500
1               Judwaa 2  1436477500
2                Ghajini  1425820000
3       Jab Tak Hai Jaan  1322642500
4               2 States  1279900000

###### 24. Find the top 5 least profitable (most loss-making) movies

In [31]:
df = pd.read_sql_query(" select movie_name, revenue - budget as loss from movies order by loss asc limit 5 ",mydb)
df

movie_name       loss
0  Bombay Velvet -748635000
1  Broken Horses -589870000
2         Mirzya -495335000
3   Jagga Jasoos -441427500
4        Zanjeer -378525000

Movie Among the listed movies, "Jab Tak Hai Jaan" had the highest revenue, "Bahubali - The Beginning" had the highest budget, "Badrinath Ki Dulhania" had the highest profit, and "Bombay Velvet" had the highest loss.

### CONCLUSION  :
    

The film industry shows strong financial performance, with Drama having the most movies, Masala Genre leading in average revenue, and "Jab Tak Hai Jaan" as the highest-grossing film. Influential figures like Akshay Kumar and Prabhas impact revenue, while directors like Ram Gopal Verma and S.S. Rajamouli affect budgets. Regular release periods are common, and original movies without remakes or franchises perform well. Introducing new talent is linked to lower budgets and revenues. Notable films include "Jab Tak Hai Jaan" for revenue, "Bahubali - The Beginning" for budget, "Badrinath Ki Dulhania" for profit, and "Bombay Velvet" for loss.